tests/src/lib.rs

```diff
@@ -457,28 +459,33 @@ mod utils {
         }
     }
 
+    /// A service deployed to the kubernetes cluster.
+    /// Service is meant as in "Microservice", not as in the Kubernetes resource called Service.
+    /// This includes a Deployment resource, a Service resource and optionally a Namespace.
     pub struct KubeService {
         pub name: String,
         pub namespace: String,
         pub target: String,
         _pod: ResourceGuard,
         _service: ResourceGuard,
+        _namespace: Option<ResourceGuard>,
     }
 
     /// randomize_name: should a random suffix be added to the end of resource names? e.g.
     ///                 for `echo-service`, should we create as `echo-service-ybtdb`.
     /// delete_after_fail: delete resources even if the test fails.
     #[fixture]
     pub async fn service(
-        #[future] kube_client: Client,
         #[default("default")] namespace: &str,
         #[default("NodePort")] service_type: &str,
         #[default("ghcr.io/metalbear-co/mirrord-pytest:latest")] image: &str,
         #[default("http-echo")] service_name: &str,
         #[default(true)] randomize_name: bool,
         #[default(false)] delete_after_fail: bool,
+        #[future] kube_client: Client,
```

Comment:
Yeah, because that way you can use the `service` fixture with injected parameters for the first arguments, and the default values for the rest. And since you're probably never going to want to specify a custom `kube_client`, it's a lot easier to use when it's the last argument instead of the first.

Now you can write a test, and let's say you want to use our normal test service, with the only different of it being in the non-default namespace, you can just use it in your test like this:

```Rust
    #[rstest]
    pu fn my_test(
        #[future] #[with("my-test-namespace")] service: KubeService,
    ) {
        ...
```

Maybe there is also a better order for the other arguments, but I think it's definitely best to have `kube_client` at the end.

Reply:
Makes sense. Helpful refactors are a plus in my books (not that I have books to keep track stuff like this).